crypto/init.c

```diff
@@ -474,12 +474,12 @@ void OPENSSL_cleanup(void)
                     "RAND_cleanup()\n");
 
 #endif
-    CRYPTO_cleanup_all_ex_data();
-    EVP_cleanup();
-    CONF_modules_free();
 #ifndef OPENSSL_NO_ENGINE
```

Comment:
Perhaps add a comment that the ordering here is fragile; test for leaks before changing things are adding things.